src/WebJobs.Script.WebHost/WebScriptHostExceptionHandler.cs

```diff
@@ -53,12 +53,8 @@ public async Task OnTimeoutExceptionAsync(ExceptionDispatchInfo exceptionInfo, T
             if (!functionInvocationDispatcher.State.Equals(FunctionInvocationDispatcherState.Default))
             {
                 _logger.LogWarning($"A function timeout has occurred. Restarting worker process executing invocationId '{timeoutException.InstanceId}'.", exceptionInfo.SourceException);
-                bool result = await functionInvocationDispatcher.RestartWorkerWithInvocationIdAsync(timeoutException.InstanceId.ToString());
-                if (!result)
-                {
-                    _logger.LogWarning($"Restarting all language worker processes since invocation Id '{timeoutException.InstanceId}' was not found.", exceptionInfo.SourceException);
-                    await functionInvocationDispatcher.RestartAllWorkersAsync();
-                }
```

Comment:
The other reason the invocation id could not be found is the worker could have completed invocation after the timeout period. This is why we were opting to restart all the workers..

Reply:
Even in that case, since invocation completed it is safe not to restart all the workers right?

Reply:
But the possibly problematic worker just happens to stay around accepting more and more invocations... I feel like if we're going down the route of just tackling one worker from a single point of failure, it might be worthwhile to revisit this approach?

[https://github.com/Azure/azure-webjobs-sdk/pull/2458](https://github.com/Azure/azure-webjobs-sdk/pull/2458)

This way, if we're timing out, the worker itself can just react to it, and we don't have to do any guess work by restarting all workers

---
In reply to: [428785504](https://github.com/Azure/azure-functions-host/pull/6072#discussion_r428785504) [](ancestors = 428785504)

Reply:
>  problematic worker

Function timeout does not indicate problematic worker. This is an enforcement on consumption plan. It is still safe for the worker to continue accepting invocations.